### PR TITLE
Add various undocumented functions from ftplib

### PR DESCRIPTION
**Pull request by PeterJCLaw**
Fixes https://github.com/python/typeshed/issues/4632

### CODE REVIEW

File: stdlib/2and3/ftplib.pyi [hunk 1]
```diff
@@ -150,3 +150,9 @@ if sys.version_info < (3,):
         def get_account(self, host: Text) -> Tuple[Optional[str], Optional[str], Optional[str]]: ...
         def get_macros(self) -> List[str]: ...
         def get_macro(self, macro: Text) -> Tuple[str, ...]: ...
+
+def parse150(resp: str) -> Tuple[str, int]: ...  # undocumented
+def parse227(resp: str) -> Tuple[str, int]: ...  # undocumented
+def parse229(resp: str, peer: Tuple[str]) -> Tuple[str, int]: ...  # undocumented
```
Comment (PeterJCLaw) [inline stdlib/2and3/ftplib.pyi:156]: Not sure about the type of `peer` here. The only usages I'm aware of pass it the result of `socket.getpeername()`, which is annotated as returning `Any`. In practise `parse229` uses the `[0]`th element from the value given as the return value for `host`, hence this annotation.

Ideally I think we'd annotate this as "tuple with `str` first element, the rest doesn't matter", not that that's something the type system can express.

Would this be better as `Any`?

File: stdlib/2and3/ftplib.pyi [hunk 2]
```diff
@@ -150,3 +150,9 @@ if sys.version_info < (3,):
         def get_account(self, host: Text) -> Tuple[Optional[str], Optional[str], Optional[str]]: ...
         def get_macros(self) -> List[str]: ...
         def get_macro(self, macro: Text) -> Tuple[str, ...]: ...
+
+def parse150(resp: str) -> Tuple[str, int]: ...  # undocumented
```
Comment (srittau) [inline stdlib/2and3/ftplib.pyi:154]: As far as I can see, this returns `Optional[int]` (at least in Python 3.7).

File: stdlib/2and3/ftplib.pyi (hunk 1 above)
Comment (srittau) [reply stdlib/2and3/ftplib.pyi:156]: I think using `Any` is the most pragmatic here. The "best" we could do with the current type system is to create a custom protocol that requires `__getitem__(self, __x: int) -> Any: ...`, but that would be complete overkill for this function.

File: stdlib/2and3/ftplib.pyi [hunk 3]
```diff
@@ -150,3 +150,9 @@ if sys.version_info < (3,):
...
+def parse227(resp: str) -> Tuple[str, int]: ...  # undocumented
+def parse229(resp: str, peer: Tuple[str]) -> Tuple[str, int]: ...  # undocumented
+def parse257(resp: str) -> Tuple[str, int]: ...  # undocumented
```
Comment (srittau) [inline stdlib/2and3/ftplib.pyi:157]: Returns a `str`.

File: stdlib/2and3/ftplib.pyi [hunk 4]
```diff
@@ -150,3 +150,9 @@ if sys.version_info < (3,):
...
+def ftpcp(source: FTP, sourcename: str, target: FTP, targetname: str = ..., type: str = ...) -> None: ...  # undocumented
```
Comment (srittau) [inline stdlib/2and3/ftplib.pyi:158]: If I understand FTP correctly, `type` can only be `"I"` or `"A"`, so we could use `Literal["A", "I"]` here.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks, remarks below.

File: stdlib/2and3/ftplib.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stdlib/2and3/ftplib.pyi:156]: I agree. Also `Tuple[str]` is definitely wrong—that means a 1-tuple containing only a str.

File: stdlib/2and3/ftplib.pyi [hunk 5]
```diff
@@ -3,7 +3,23 @@ from _typeshed import SupportsRead, SupportsReadline
...
-from typing import Any, BinaryIO, Callable, Dict, Iterable, Iterator, List, Optional, Text, TextIO, Tuple, Type, TypeVar, Union
+from typing import (
+    Any,
+    BinaryIO,
+    Callable,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Literal,
```
Comment (srittau) [inline stdlib/2and3/ftplib.pyi:14]: Unfortunately, we have to import `Literal` from `typing_extensions`, as not all version of `typing` have this.

**Review by srittau — APPROVED** [review]

**Pull request merged**